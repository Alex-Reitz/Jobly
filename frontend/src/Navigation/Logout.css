.exit-message {
  font-size: 1.5em;
  padding: 10px;
}

.buttonFalse {
  background-color: #316685;
  border: none;
  color: white;
  padding: 10px;
  text-decoration: none;
  margin: 4px 2px;
  font-size: 1rem;
  cursor: pointer;
  border-radius: 5px;
}

.logout-container {
  margin: 0;
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
}

.buttonTrue {
  background-color: white;
  border: none;
  color: black;
  padding: 10px;
  text-decoration: none;
  margin: 4px 2px;
  font-size: 1rem;
  cursor: pointer;
  border-radius: 5px;
}
