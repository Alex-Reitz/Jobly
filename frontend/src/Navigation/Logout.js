import React from "react";
import "./Logout.css";

function Logout({ logout }) {
  return (
    <div className="logout-container">
      <h4 className="exit-message">
        Click logout to leave. We'll see you next time!
      </h4>
      <button id="logout" onClick={logout}>
        Logout
      </button>
    </div>
  );
}

export default Logout;
