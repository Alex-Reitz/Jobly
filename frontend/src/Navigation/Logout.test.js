import React from "react";
import { render } from "@testing-library/react";
import Logout from "./Logout";
import { MemoryRouter } from "react-router";
import UserContext from "../Context/userContext";

it("matches snapshot", function () {
  const testFunc = jest.fn();
  const { asFragment } = render(
    <MemoryRouter>
      <UserContext.Provider value={{ currentUser: testFunc }}>
        <Logout />
      </UserContext.Provider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
