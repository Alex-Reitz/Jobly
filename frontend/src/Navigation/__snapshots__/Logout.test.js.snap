// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="logout-container"
  >
    <h4
      class="exit-message"
    >
      Click logout to leave. We'll see you next time!
    </h4>
    <button
      class="buttonFalse"
      id="logout"
    >
      Logout
    </button>
  </div>
</DocumentFragment>
`;
